Kinetic_space_partition/include/CGAL/Kinetic_space_partition_3.h

```diff
@@ -0,0 +1,2395 @@
+    return m_volumes.size();
+  }
+
+  /*!
+  \brief provides the support planes of the partition derived from the input polygons
+
+  @return
+   vector of planes.
+
+  \pre inserted polygons
+  */
+  const std::vector<typename Intersection_kernel::Plane_3> &input_planes() const {
+    return m_input_planes;
+  }
+
+  /*!
+   \brief exports the kinetic partition into a `Linear_cell_complex_for_combinatorial_map<3, 3>` using a model of `KineticLCCItems` as items, e.g., `Kinetic_space_partition_3::Linear_cell_complex_min_items`.
+
+   Volume and face attributes defined in the model `KineticLCCItems` are filled. The volume index is in the range [0, number of volumes -1]
+
+   \tparam LCC must be a model of `Linear_cell_complex_for_combinatorial_map<3, 3>` using a model of `KineticLCCItems`.
+
+   \param lcc instance of LCC to be filled with the kinetic partition. Any data contained in `lcc` will be cleared before.
+
+   \pre created partition
+  */
+  template<class LCC>
+  void get_linear_cell_complex(LCC &lcc) const {
+    lcc.clear();
+
+    std::map<Index, std::size_t> mapped_vertices;
+    std::map<typename Intersection_kernel::Point_3, std::size_t> mapped_points;
+    std::vector<typename Intersection_kernel::Point_3> vtx;
+    std::vector<Index> vtx_index;
+
+    From_exact to_inexact;
+    To_exact to_exact;
+
+    std::vector<Index> faces_of_volume, vtx_of_face;
+    std::vector<typename Intersection_kernel::Point_3> pts_of_face;
+
+    for (std::size_t i = 0; i < number_of_volumes(); i++) {
+      faces(i, std::back_inserter(faces_of_volume));
+
+      for (const Index& f : faces_of_volume) {
+        exact_vertices(f, std::back_inserter(pts_of_face), std::back_inserter(vtx_of_face));
+
+        for (std::size_t j = 0; j < pts_of_face.size(); j++) {
+          auto pit = mapped_points.emplace(pts_of_face[j], vtx.size());
+          if (pit.second) {
+            mapped_vertices[vtx_of_face[j]] = vtx.size();
+            vtx.push_back(pts_of_face[j]);
+            vtx_index.push_back(vtx_of_face[j]);
+          }
+          else mapped_vertices[vtx_of_face[j]] = pit.first->second;
+        }
+
+        pts_of_face.clear();
+        vtx_of_face.clear();
+      }
+      faces_of_volume.clear();
+    }
+
+    CGAL::Linear_cell_complex_incremental_builder_3<LCC> ib(lcc);
+    for (std::size_t i = 0; i < vtx.size(); i++)
+      ib.add_vertex(vtx[i]);
+
+    std::size_t num_faces = 0;
+    //std::size_t num_vols = 0;
+    //std::size_t num_vtx = 0;
+
+    typename LCC::Dart_descriptor d;
+
+    std::vector<bool> used_vertices(mapped_vertices.size(), false);
+    std::vector<bool> added_volumes(number_of_volumes(), false);
+    std::deque<std::size_t> queue;
+    queue.push_back(0);
+    while (!queue.empty()) {
+      std::size_t v = queue.front();
+      queue.pop_front();
+
+      if (added_volumes[v])
+        continue;
+
+      if (!can_add_volume_to_lcc(v, added_volumes, mapped_vertices, used_vertices)) {
+        queue.push_back(v);
```

Comment:
The infinite loop in the testsuite is here.  We have 47 elements in the queue which all can not be added to the LCC.